dependencies.gradle

```diff
@@ -1,43 +1,43 @@
 def versions = [
     kotlinVersion                : '1.3.72',
```

Comment:
this kotlin version is fairly out of date, the newer versions have made changes to unsigned number handling and I think it would be good to get those updates in as you do so much unsigned manipulation

Reply:
I tried 1.5 and 1.4 but ran into issues that i did not want to deal with, I can give it another try.